sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosStoredProcedureResponse.java

```diff
@@ -24,7 +28,7 @@
     }
 
     CosmosStoredProcedureResponse(StoredProcedureResponse response) {
-        super(response);
```

Comment:
Can you try and see if this works?
```
    super(new ResourceResponse<>(response.getRxDocumentServiceResponse(),
                                     StoredProcedure.class));
```
and in StoredProcedureResponse
```
    public RxDocumentServiceResponse getRxDocumentServiceResponse() {
        return this.response;
    }
```

In this way you may avoid re implementing all the super methods

Reply:
Thanks @mbhaskar for the suggestion, I have implemented it, will run full CI to verify it.